### PR TITLE
Fix lockfile usage - Closes #529

### LINKED ISSUE #529: Lockfile with getStdin on non interactive mode does not work
### Description
Lockfile approach is not compatible with stdin on NON_INTERACTIVE_MODE
```
if (process.env.NON_INTERACTIVE_MODE) {
	const stdin = await getStdIn({ dataIsRequired: true });
	return stdin.data;
}
```

```
./bin/lisky create transaction 0 200 123L --no-signature --json | ./bin/lisky sign transaction
Config lockfile at /Users/will/.lisky/config.lock found. Are you running Lisky in another process?
╔══════════════════════════════════════════════════════════╗
║ error                                                    ║
╟──────────────────────────────────────────────────────────╢
║ Could not sign transaction: No transaction was provided. ║
╚══════════════════════════════════════════════════════════╝
```

### Steps to Reproduce

```
./bin/lisky create transaction 0 200 123L --no-signature --json | ./bin/lisky sign transaction
```

**Expected behavior:** [What you expect to happen]
It should be able to cast stdin to next command

**Actual behavior:** [What actually happens]
Throws an error

**Reproduces how often:** [What percentage of the time does it reproduce?]

### Versions
1.0.0

### PR DESCRIPTION
**Pull request by shuse2**
### What was the problem?
When in non interactive mode, it was unable to pipe commands because of the lockfile.

### How did I fix it?
Only use lockfile when it needs to.
In this case, only when writing to config file, it is locked.

### How to test it?
```
./bin/lisky create transaction 0 200 123L --no-signature --json | ./bin/lisky sign transaction
```
and check if there is no error.
### Review checklist

* The PR solves #529 
* All new code is covered with unit tests
* All new code was formatted with Prettier
* Linting passes
* Tests pass
* Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
* Documentation has been added/updated

### CODE REVIEW

File: test/specs/utils/api.js [hunk 1]
```diff
@@ -13,120 +13,104 @@
...
 import * as given from '../../steps/1_given';
 import * as when from '../../steps/2_when';
 import * as then from '../../steps/3_then';
 
 describe('API util', () => {
+	beforeEach(setUpUtilAPI);
 	Given(
-		'a config with "api.network" set to "main"',
-		given.aConfigWithAPINetworkSetTo,
+		'a config with api.network and api.nodes set to "main" and "http://localhost:4000"',
```
Comment (willclarktech) [inline test/specs/utils/api.js]: I think `'a config with api.network set to "main" and api.nodes set to "http://localhost:4000"'` is less confusing.

File: test/specs/utils/api.js [hunk 2]
```diff
@@ -13,120 +13,104 @@
...
+		given.aConfigWithAPINetworkAndNodesSetTo,
```
Comment (willclarktech) [inline test/specs/utils/api.js]: `aConfigWithAPINetworkAndAPINodesSetTo`

File: test/specs/utils/config.js [hunk 3]
```diff
@@ -22,18 +22,6 @@ describe('config util', () => {
 	beforeEach(setUpUtilConfig);
 	afterEach(tearDownUtilConfig);
 	Given('a default config', given.aDefaultConfig, () => {
-		Given(
```
Comment (willclarktech) [inline test/specs/utils/config.js]: Do we not need this any more?

File: test/specs/utils/config.js [hunk 4]
```diff
@@ -136,8 +108,8 @@ describe('config util', () => {
 										given.thereIsAConfigLockfile,
 										() => {
 											When(
-												'the config is loaded',
-												when.theConfigIsLoaded,
+												'setConfig is called',
```
Comment (willclarktech) [inline test/specs/utils/config.js]: Now we're mixing and matching `setConfig` and `getConfig`. Do we want two different paths?

File: test/specs/utils/helpers.js [hunk 5]
```diff
@@ -561,6 +562,7 @@ describe('utils helpers', () => {
 		);
 	});
 	describe('#shouldUseJSONOutput', () => {
+		beforeEach(setUpUtilHelpers);
```
Comment (willclarktech) [inline test/specs/utils/helpers.js]: Prefer doing this at the outermost level, or having individual set up functions.

File: test/steps/config/3_then.js [hunk 6]
```diff
@@ -68,12 +74,12 @@ export function itShouldResolveToTheConfig() {
 	return expect(returnValue).to.eventually.eql(config);
 }
 
-export function theDefaultConfigShouldBeExported() {
+export function theDefaultConfigShouldBeReturned() {
```
Comment (willclarktech) [inline test/steps/config/3_then.js]: This function seems not to be used.

File: test/steps/config/1_given.js [hunk 7]
```diff
@@ -37,61 +37,58 @@ export function aConfig() {
...
 	this.test.ctx.config = config;
 }
 
 export function thereIsAConfigLockfile() {
-	const error = new Error('Found a lockfile');
-	lockfile.lock.throws(error);
+	lockfile.checkSync.returns(true);
 }
 
 export function thereIsNoConfigLockfile() {}
```
Comment (willclarktech) [inline test/steps/config/1_given.js]: The final function in this file is not used currently.

File: test/steps/files/3_then.js [hunk 8]
```diff
@@ -78,3 +71,11 @@ export function itShouldResolveToTheFirstLineOfTheFile() {
 	const { returnValue, passphrase } = this.test.ctx;
 	return expect(returnValue).to.eventually.equal(passphrase);
 }
+
+export function itShouldLockTheFile() {
+	return expect(lockfile.lockSync).to.be.called;
+}
+
+export function itShouldUnkockTheFile() {
```
Comment (willclarktech) [inline test/steps/files/3_then.js]: typo: `Unlock`

File: src/utils/config.js [hunk 9]
```diff
@@ -83,7 +86,22 @@ const attemptToValidateConfig = (config, path) => {
 	return attemptCallWithError(fn, errorMessage);
 };
 
-const getConfig = () => {
+export const setConfig = newConfig => {
+	if (!process.env.EXEC_FILE_CHILD) {
```
[the remark this answers is not in the document]
Comment (shuse2) [reply src/utils/config.js]: yes we don't need it anymore

File: test/specs/utils/config.js (hunk 4 above)
Comment (shuse2) [reply test/specs/utils/config.js]: I thought this config tests are mostly based on setup rather than function, so I followed the flow.

**Pull request merged**